### PR TITLE
Complete Business Details backend. Excludes reviews backend.

### PR DESCRIPTION
**Pull request by saraholiviawu**
**Description**: Complete Business Details page backend. Reads data from Firestore and localStorage to display on the Business Details page via JQuery and the DOM. Excludes reviews feature and features to be owned by Smruthi (image gallery, business tags). 

**Key features broken by section**:
 - **Maps with walking directions, duration, and distance (Maps JavaScript API Directions service)**:
     - Renders a map initialized to the the center of the United States. Read user address from localStorage; set it as the travel route origin. Read business address from Firestore and set it as the travel route destination. Calculate and display the route.
     - Calculate the midpoint of the route and open an info window containing a walking icon, walking duration, and walking distance. These values were calculated once already in the main page / business gallery page and should not be calculated using the Distance matrix a second time. Therefore, these values have been set in localStorage from the first calculation and are pulled here.
  - **Quick information section**: 
    - **Read Firebase Firestore and localStorage data**: Read and output business street address, user's walking distance from business, business phone number, and business website containing an embedded href redirect link. Read business open hours range.  
    - **Open or closed status**: Display "Open" or "Closed" in the Quick Info section depending on whether current hour and minute is within the business open hours range for this particular day of the week (starting time is inclusive, ending time is exclusive). If business is not open at all today, display "Closed."
  - **User reviews summary**:   
    - **Calculate star and price ratings**: See PR #34 for details
  - **Add a review**: Pop up form records star and price ratings in local storage. See PR #34. Backend WIP.
  -  **Latest Announcements**: Firestore reads
     - **Timestamp**: Read timestamp object from Firestore, convert to a date string, parse from "Wed Jul 01 2020 12:00:00 GMT-0700 (Pacific Daylight Time)" to "July 1, 2020". 
  - **Our story**: Firestore reads
  - **Our team**:  Firestore reads
    - **Serve blobs** representing the avatar image for each team member. 
     - **Add an onclick function** to the image that contains the team member's name, position, and bio in the parameter. Generate unique ids to access during onclick function. Onclick function unhighlights all team member avatars and highlights the team member of interest. Swap the existing team member info. 
        - Use regex to escape special characters (, ), ', for parameters in the onclick function that interfere with onclick functionality. 
        - Replace double quote " with two single quotes ' b/c the former cannot be escaped properly in output onto DOM
        - This parse function is placed in the read since the time complexity and memory for this operation is not significant. Parsing the data on the write within managebusiness.html would instead would require adding AND removing special characters in the managebusiness.html.
  - **Our safety health commitment**:  
     - **Determine blob, text, and color to add**: Read and iterate through list of values from Firestore corresponding to whether a service "offered" or "not offered"
  - **Open Hours**: Read from Firestore to display each day of the week's open hours.
     - **Highlight today's hours**: Get current day of the week and highlight today's hours. Will still be highlighted even if current hour and minute is out of today's open hours range  
  - **Order delivery**: Read from Firestore, iterate through each delivery link, and add it as an href for each corresponding delivery logo image (Uber Eats, Postmates, Grubhub). If the link contains empty string, hide the delivery image. 

**Demo**:
1. Full page display
BD Backend Complete Page 1
BD Backend Complete Page 2
BD Backend Complete Page 3

2. Our team section: Activate onclick function that swaps displayed team member's name, position, bio 
BD Our team

3. Delivery Links: If postmates link is not specified, hide image logo
BD No specified postmates link

### CODE REVIEW

File: ripple/src/main/webapp/maps.js [hunk 1]
```diff
@@ -6,39 +6,95 @@ script.defer = true;
...
+            zoomControl: true,
+          });
+          directionsRenderer.setMap(map);
+          calculateAndDisplayRoute(map, directionsService, directionsRenderer, destination);
+
+        } else {
+          console.log("Error: business document does not exist");
+        }
+      }
+  getDocByDocId("businesses", businessId, lambda);
```
Comment (sabadas) [inline ripple/src/main/webapp/maps.js:61]: maybe give a more descriptive name for lambda?

File: ripple/src/main/webapp/businessdetails.js [hunk 2]
```diff
@@ -219,7 +256,291 @@ function calculatePrice(avgPriceRatings, numPriceRatings) {
   }
 }
 
-/* Display the number of dollar signs for price rating on the page */
+/* Add a review: Display the number of dollar signs for price rating on the page */
```
Comment (sabadas) [inline ripple/src/main/webapp/businessdetails.js:259]: not sure how i understand price is related to review. Was it intentional to add to all the comments?

File: ripple/src/main/webapp/businessdetails.js [hunk 3]
```diff
@@ -197,14 +234,14 @@ function displayStarReviewSummary(avgStarRatings, numStarRatings) {
...
-/* Given an float, function rounds decimals to the tenths place. Ex: 3.14 -> 3.1 */
+/* Add a review: Given an float, function rounds decimals to the tenths place. Ex: 3.14 -> 3.1 */
 function RoundDecimals(float) {
   var roundedFloat = Math.round(float * 10) / 10;
   return roundedFloat;
   console.log("roundedFloat: " + roundedFloat);
 }
 
-/* If no reviews yet, default price display is N/A. 
+/* Add a review: If no reviews yet, default price display is N/A.
```
Comment (sabadas) [inline ripple/src/main/webapp/businessdetails.js:244]: not sure how i understand price is related to review

File: ripple/src/main/webapp/businessdetails.js (hunk 2 above)
Comment (saraholiviawu) [reply ripple/src/main/webapp/businessdetails.js:259]: Yes it was intentional. When you add a review, you're rating by stars and by price

File: ripple/src/main/webapp/businessdetails.js (hunk 3 above)
Comment (saraholiviawu) [reply ripple/src/main/webapp/businessdetails.js:244]: See comment above. These are functions that are all called when a user adds a review.

File: ripple/src/main/webapp/maps.js (hunk 1 above)
Comment (saraholiviawu) [reply ripple/src/main/webapp/maps.js:61]: Fixed.

File: ripple/src/main/webapp/businessdetails.js (hunk 3 above)
Comment (saraholiviawu) [reply ripple/src/main/webapp/businessdetails.js:244]: Will rename. See comment above

File: ripple/src/main/webapp/businessdetails.js (hunk 2 above)
Comment (saraholiviawu) [reply ripple/src/main/webapp/businessdetails.js:259]: Hmm actually, it's more for displaying a review summary of all users. Will change.

**Pull request merged**